src/confluent_kafka/schema_registry/avro.py

```diff
@@ -185,23 +185,17 @@ class AvroSerializer(Serializer):
                      'subject.name.strategy': topic_subject_name_strategy}
 
     def __init__(self, schema_registry_client, schema_str, to_dict=None, conf=None):
-        if isinstance(schema_str, str):
-            schema = _schema_loads(schema_str)
-        elif isinstance(schema_str, Schema):
-            schema = schema_str
-        else:
+        if not isinstance(schema_str, (str, Schema)):
             raise TypeError('You must pass either schema string or schema object')
 
+        self._to_dict = to_dict
```

Comment:
I will move the two validation IFs to the start of the method. We still need `self._to_dict = to_dict` because it is an instance variable required on other class methods